### PR TITLE
8 - split dynamic import and dynamic require

### PR DESCRIPTION
**Pull request by unicop**
# motivation

The ESLint options are not that intuitive for certain rules (including this one), and we aim to give a VERY GOOD experience in comparison to the existing plugin available.

Split into 2 rules 
- no-commonjs-dynamic-require
- no-esm-dynamic-import

### CODE REVIEW

**Comment by unicop** [comment]
[[rule] split no-dynamic-import to no-dynamic-require/import (2 rules) for simplicity](https://trello.com/c/qIGoaPO0/8-rule-split-no-dynamic-import-to-no-dynamic-require-import-2-rules-for-simplicity)

File: src/rules/no-commonjs-dynamic-require.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import { TSESTree, AST_NODE_TYPES } from '@typescript-eslint/types';
+
+import { createRule } from '../util';
+
+export type Options = [{ }];
```
Comment (neriyarden) [inline src/rules/no-commonjs-dynamic-require.ts:5]: Toptions?

File: src/rules/no-commonjs-dynamic-require.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+function isStaticValue(arg: TSESTree.CallExpressionArgument | TSESTree.Expression) {
+    return arg.type === AST_NODE_TYPES.Literal ||
+        (arg.type === AST_NODE_TYPES.TemplateLiteral && arg.expressions.length === 0);
+}
+
+
+export default createRule<Options, MessageIds>({
+    defaultOptions: [{} = {}],
+    name: 'no-dynamic-import',
```
Comment (neriyarden) [inline src/rules/no-commonjs-dynamic-require.ts:24]: change name of rule

File: src/rules/no-commonjs-dynamic-require.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        },
+        messages: {
+            requireShouldBeLiteral: 'Calls to require() should use string literals'
+        },
+        schema: [{}],
+    },
+
+    create(context) {
+
+        const [{ } = { }] = context.options;
```
Comment (neriyarden) [inline src/rules/no-commonjs-dynamic-require.ts:40]: what does the [{ } = { }] mean?

**Review by neriyarden — APPROVED** [review]

File: src/rules/no-commonjs-dynamic-require.ts (hunk 1 above)
Comment (unicop) [reply src/rules/no-commonjs-dynamic-require.ts:5]: soon these issues will be blocked :) 
we can use .github action `reviewdog` to comment linting issues

**Pull request merged**